pkg/roachprod/vm/azure/azure.go

```diff
@@ -645,11 +645,11 @@ func (p *Provider) List(l *logger.Logger, opts vm.ListOptions) (vm.List, error)
 			RemoteUser:  remoteUser,
 			VPC:         "global",
 			MachineType: string(found.HardwareProfile.VMSize),
-			CPUArch:     CpuArchFromAzureMachineType(string(found.HardwareProfile.VMSize)),
 			// We add a fake availability-zone suffix since other roachprod
 			// code assumes particular formats. For example, "eastus2z".
 			Zone: *found.Location + "z",
 		}
+		m.CPUArch, _ = p.GetVMArchitecture(l, &m)
```

Comment:
Should we maybe add the appropriate error handling here, so that if someone implements it in the future when Azure does start exposing architecture, it won't hide errors?